lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
+  @moduledoc false
+
+  alias Screens.V2.Departure
+  alias Screens.V2.ScreenData.QueryParams
+  alias Screens.V2.WidgetInstance.Departures, as: DeparturesWidget
+  alias Screens.V2.WidgetInstance.Departures.NormalSection
+  alias Screens.V2.WidgetInstance.{DeparturesNoData, DeparturesNoService}
+  alias ScreensConfig.Screen
+
+  @max_departure_results 2
+
+  @type widget ::
+          DeparturesNoData.t() | DeparturesNoService.t() | DeparturesWidget.t()
+  @type options :: [
+          departure_fetch_fn: Departure.fetch(),
+          now: DateTime.t()
+        ]
+
+  @spec departures_candidate(Screen.t(), QueryParams.t(), options()) :: [widget()]
+  def departures_candidate(config, query_params, options \\ [])
+
+  def departures_candidate(
+        config,
+        %{stop_id: stop_id, route_id: route_id},
+        options
+      ) do
+    now = Keyword.get(options, :now, DateTime.utc_now())
+    fetch_fn = Keyword.get(options, :departure_fetch_fn, &Departure.fetch/2)
+
+    route_id
+    |> fetch_departures(stop_id, fetch_fn)
+    |> process_response(config, now)
+  end
+
+  def departures_candidate(config, _query_params, _options) do
+    build_no_data_widget(config)
+  end
+
+  defp fetch_departures(route_id, stop_id, departure_fetch_fn) do
+    fetch_params = %{:stop_ids => [stop_id]}
+    fetch_opts = [include_schedules: false]
+
+    with {:ok, departures} <- departure_fetch_fn.(fetch_params, fetch_opts) do
+      {:ok,
+       departures
+       |> filter_current_route(route_id)}
+    end
+  end
+
+  # Only return departures that are not from the bus's current route in either direction
+  defp filter_current_route(departures, route_id) do
+    Enum.filter(departures, &(&1.prediction.route.id != route_id))
+  end
+
+  @spec process_response({:error, any()}, Screen.t(), DateTime.t()) :: [DeparturesNoData]
+  @spec process_response({:ok, []}, Screen.t(), DateTime.t()) :: [DeparturesNoService]
+  @spec process_response({:ok, [Departure.t()]}, Screen.t(), DateTime.t()) :: [DeparturesWidget]
+  defp process_response({:error, _}, config, _) do
+    build_no_data_widget(config)
+  end
+
+  defp process_response({:ok, []}, config, _) do
+    [%DeparturesNoService{screen: config}]
+  end
+
+  defp process_response({:ok, departure_data}, config, now) do
+    [
+      %DeparturesWidget{
+        screen: config,
+        sections: [
+          %NormalSection{
+            rows: departure_data,
+            layout: %ScreensConfig.V2.Departures.Layout{
+              base: nil,
+              include_later: false,
+              max: @max_departure_results,
```

Comment:
Since the design has a fixed number of departures rather than being able to flex (like the multiple sections on Sectionals, which are the reason the Layout parameters exist), I think this could be better implemented as a limit on the backend, e.g. using `Enum.take`. Then we won't send a potentially large number of departures to the frontend only to throw most of them away.

Reply:
Agreed this should be on the BE! I was thinking that could be done in the follow up filtering story, but it's so quick that I might as well add it now.